go/arrow/flight/flightsql/example/sqlite_info.go

```diff
@@ -0,0 +1,196 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build go1.17
+// +build go1.17
+
+package example
+
+import "github.com/apache/arrow/go/v10/arrow/flight/flightsql"
+
+func SqlInfoResultMap() flightsql.SqlInfoResultMap {
+	return flightsql.SqlInfoResultMap{
+		uint32(flightsql.SqlInfoFlightSqlServerName):         "db_name",
+		uint32(flightsql.SqlInfoFlightSqlServerVersion):      "sqlite 3",
+		uint32(flightsql.SqlInfoFlightSqlServerArrowVersion): "10.0.0-SNAPSHOT",
```

Comment:
when built it with go1.18, there is fancy new buildinfo information added to the binary which detects the VCS information and would provide us the version. Sadly, we're still tied to go1.16. The standard for Go version handling is N-2 where N is the latest. In this case, N would be go1.19 (we could bump to go1.17, but there isn't really a need to). When go1.20 comes out (roughly 5 - 6 months) then it would be good for us to upgrade to go1.18 and take advantage of several things such as the new buildinfo and generics.

Until then, I could add something in the Arrow library to expose version information like I did with the parquet libraries version info.